packages/python/plotly/plotly/express/_core.py

```diff
@@ -754,6 +754,215 @@ def apply_default_cascade(args):
+        argument_list = (
+            [args.get(field_name)]
+            if field_name not in array_attrables
+            else args.get(field_name)
+        )
+        # argument not specified, continue
+        if argument_list is None or argument_list is [None]:
+            continue
+        # Argument name: field_name if the argument is not a list
+        # Else we give names like ["hover_data_0, hover_data_1"] etc.
+        field_list = (
+            [field_name]
+            if field_name not in array_attrables
+            else [field_name + "_" + str(i) for i in range(len(argument_list))]
+        )
+        # argument_list and field_list ready, iterate over them
+        # Core of the loop starts here
+        for i, (argument, field) in enumerate(zip(argument_list, field_list)):
+            length = len(df)
+            if argument is None:
+                continue
+            ## ----------------- argument is a col name ----------------------
+            elif isinstance(argument, str) or isinstance(
+                argument, int
+            ):  # just a column name given as str or int
+                if not df_provided:
+                    raise ValueError(
+                        "String or int arguments are only possible when a"
+                        "DataFrame or an array is provided in the `data_frame`"
+                        "argument. No DataFrame was provided, but argument '%s'"
+                        "is of type str or int." % field
+                    )
+                # Check validity of column name
+                if argument not in args["data_frame"].columns:
+                    err_msg = (
+                        "Value of '%s' is not the name of a column in 'data_frame'. "
+                        "Expected one of %s but received: %s"
+                        % (field, str(list(args["data_frame"].columns)), argument)
+                    )
+                    if argument == "index":
+                        err_msg += (
+                            "\n To use the index, pass it in directly as `df.index`."
+                        )
+                    raise ValueError(err_msg)
+                if length and len(args["data_frame"][argument]) != length:
+                    raise ValueError(
+                        "All arguments should have the same length. "
+                        "The length of column argument `df[%s]` is %d, whereas the "
+                        "length of previous arguments %s is %d"
+                        % (
+                            field,
+                            len(args["data_frame"][argument]),
+                            str(list(df.columns)),
+                            length,
+                        )
+                    )
+                col_name = argument
+                if isinstance(argument, int):
+                    col_name = _name_heuristic(argument, field, reserved_names)
+                    if field_name not in array_attrables:
```

Comment:
could we remove this `if/else` block as well as the `continue` and let it "fall through" to the bottom of the loop where we have the same logic?